### ISSUE TITLE
Référencement - Différencier balises Title et H1

### ISSUE THREAD

**Issue by babastienne** [GeotrekCE/Geotrek-rando-v3#992]
Le parc National de la Vanoise a remonté un besoin d'amélioration du référencement côté Geotrek-Rando. 

Actuellement sur la page détail d'un objet, le code est organisé ainsi : 
- Balise `Title` : il s'agit d'un texte formaté ainsi "Nom de la randonnée - Titre du site Geotrek-Rando".
- Balise `H1` : nom de la randonnée

Il y a redondance entre les deux champs et via l'usage d'un autre titre de page le référencement de GTR3 pourrait être amélioré. 

Comment au mieux développer cette amélioration ? Quelques pistes : 
- Faut-il que le titre soit le même pour toutes les pages (auquel cas cela pourrait être défini côté GTR3) ? Je ne suis pas favorable.
- Faut-il que l’utilisateur ai la main sur le titre côté Geotrek-Admin pour pouvoir décider pour chaque objet du titre associé ? 
    - Si oui cela implique d'ajouter un champ `titre` dans le modèle de données et d'adapter le formulaire pour que sur chaque objet un utilisateur puisse définir le champ
    - Il faudra bien préciser qu'il s'agit du titre de la page essentiellement utilisé pour le référencement (et nom de l'onglet), car sinon certains utilisateur ne comprendrons pas pourquoi ce titre n'est pas visible sur la page
    - Comment gérer le cas où le champ n'est pas rempli ? Faut-il garder la combinaison de titre par défaut actuel dans le cas où l'utilisateur n'a pas rempli ce champ ?
    - Il faudra ajouter à l'API de GTA ce champ puis implémenter la logique de génération automatique de titre si vide côté GTA
    - Côté GTR3 développement pour exploiter ce champ.
    - Quid des autres type de contenus (outdoor, contenus touristiques, évènement touristiques) possédants une fiche détail, faut-il également ajouter un champ `titre` sur ces objets ?
- Une autre solution ?

**Comment by camillemonchicourt** [comment]
Je ne comprends pas le soucis.
Au contraire, on a fait ça pour être bien puissant niveau SEO et maximiser le référencement sur le nom de chaque rando.

De mon côté, je ne veux surtout pas que les utilisateurs touchent à ça et le modifient car c'est déjà taillé et optimisé pour le référencement.

**Comment by dtrucs** [comment]
Je suis d'accord avec Camille, je ne vois pas vraiment ce qu'on peut faire de mieux à ce niveaux là.
Est-ce le parc National de la Vanoise peut fournir un document qui prouve que la redondance puisse pénaliser le SEO d'une manière ou d'une autre ?

**Comment by babastienne** [comment]
Pour des besoins d'améliorations de SEO, le CD61 nous demande de pouvoir personnaliser les meta description et title des pages statiques et des "catégories" des entités Trek / Touristic Content / Touristic Event / Outdoor.

L'idée serait donc d'ajouter sur les contenus suivants des champs "Meta Description" et "Meta Title" : 
- Flatpages
- Pratique (trek)
- Pratique (outdoor)
- Catégorie de contenu touristique
- Type d'événement touristiques

Ces champs seraient optionnels : si pas remplis alors on utilise la valeur par défaut générée par Rando. Sinon on surcharge avec la valeur configurée dans l'admin et renvoyée par l'API.

**Comment by dtrucs** [comment]
> Pour des besoins d'améliorations de SEO, le CD61 nous demande de pouvoir personnaliser les meta description et title des pages statiques et des "catégories" des entités Trek / Touristic Content / Touristic Event / Outdoor.
> 
> L'idée serait donc d'ajouter sur les contenus suivants des champs "Meta Description" et "Balise Title" sur GTA :
> 
>     * Flatpages
> 
>     * Pratique (trek)
> 
>     * Pratique (outdoor)
> 
>     * Catégorie de contenu touristique
> 
>     * Type d'événement touristiques
> 
> 
> Ces champs seraient optionnels : si pas remplis alors on utilise la valeur par défaut générée par Rando. Sinon on surcharge avec la valeur configurée dans l'admin et renvoyée par l'API.
> 
> Pour appuyer cette demande le CD61 se base sur un audit SEO réalisé sur leur site [Randonnée Normandie](https://randonnee-normandie.com/) par une société experte sur le sujet qui a révélé un besoin sur cet axe (et d'autres qui feront l’objet d'autres tickets).

Pour préciser la demande, sur GTR les valeurs de ces champs seront appliquées lors de la recherche de "catégorie" unique.

**Comment by camillemonchicourt** [comment]
OK intéressant.
Mais pas forcément très compréhensible pour ceux qui vont saisir.
Autant sur les pages statiques, avoir un champs "Meta Description" et "Balise Title" peut se comprendre (en les nommant mieux et avec un helptext pour expliquer), autant pour ceux qui vont créer des pratiques, ça va leur faire bizarre...
J'ai une petite réserve sur le fait d'ajouter ces champs sur les pratiques.

**Comment by babastienne** [comment]
Suite du sujet : 

Le territoire nous a demandé un devis car il souhaitait financer une évolution pour permettre de sélectionner les valeurs des champs "Meta Description" et "Balise Title" sur les contenus issus de GTA (contenus listés dans le commentaire précédent, à savoir Flatpages, Pratiques trek & outdoor, catégories contenus touristiques, types d'évènements).

Étant donné les réserves émises par la communauté sur le ticket nous avons proposé un développement uniquement pour la partie Flatpages et pas pour les autres contenus pour ne pas aller à l'encontre de l'avis émis par @camillemonchicourt.

Le territoire n'a donc pas souhaité donner suite car c'était "tout ou rien" et financer un développement uniquement pour les pages statiques n'était pas souhaitable.
